packages/webarchitect/src/executeValidationRules.ts

```diff
@@ -0,0 +1,59 @@
+import type { Schema, ValidationResult } from "./types.js";
+import validateDirectoryRule from "./validateDirectoryRule.js";
+import validateFileRule from "./validateFileRule.js";
+
+/**
+ * Executes all validation rules defined in a schema against a project.
+ * Iterates through each rule in the schema, determines whether it's a file
+ * or directory rule, and delegates to the appropriate validator.
+ *
+ * @param projectPath - Absolute or relative path to the project directory to validate
+ * @param schema - Schema object containing validation rules to execute
+ * @returns Promise that resolves to an array of validation results
+ *
+ * @example
+ * ```typescript
+ * const schema = {
+ *   name: "my-schema",
+ *   "package-config": {
+ *     file: { name: "package.json", contains: {...} }
+ *   }
+ * };
+ * const results = await executeValidationRules("/path/to/project", schema);
+ * ```
+ */
+export default async function executeValidationRules(
+  projectPath: string,
+  schema: Schema,
+): Promise<ValidationResult[]> {
+  const results: ValidationResult[] = [];
+
+  // Iterate through each rule in the schema
+  for (const [ruleName, rule] of Object.entries(schema)) {
+    // Skip meta-properties that aren't validation rules
+    if (ruleName === "$schema" || ruleName === "name" || ruleName === "extends")
+      continue;
+
+    // Type guard to ensure rule is an object with 'file' or 'directory'
+    if (typeof rule === "object" && rule !== null) {
+      if ("file" in rule) {
+        // Pass the rule name context to file validation
+        const fileResults = await validateFileRule(
+          projectPath,
+          rule.file,
+          ruleName,
+        );
+        results.push(...fileResults);
+      } else if ("directory" in rule) {
+        // Pass the rule name context to directory validation
+        const dirResults = await validateDirectoryRule(
+          projectPath,
+          rule.directory,
+          ruleName,
+        );
+        results.push(...dirResults);
+      }
+    }
```

Comment:
At the risk of abstracting too early, this looks like a decent candidate for abstraction. validateFileRule and validateDirectoryRule are very similar in structure and function. They are also called very similarly here. I could see this branching becoming a bit cumbersome and verbose in this file if more validation rule types are added in the future. 

Maybe we could add a `validateRule` function that, given a rule, calls the appropriate validator (rule, directory, any other we may come across). This would clean up this validation logic at this level and may also consolidate some of the logic in the file and directory validators.